src/app/components/search/SearchInputs.tsx

```diff
@@ -0,0 +1,108 @@
+import { Input, Option, Select, Typography } from "@mui/joy";
+import useSearch from "../../hooks/useSearch";
+import { InfoOutlined } from "@mui/icons-material";
+import { useEffect } from "react";
+
+export default function SearchInputs({
+  onUpdateCourses,
+  onUpdateLoading,
+  onUpdateLoadingMoreResults,
+}: {
+  onUpdateCourses: any;
+  onUpdateLoading: any;
+  onUpdateLoadingMoreResults: any;
+}) {
+  const {
+    courses,
+    loading,
+    loadingMoreResults,
+    departments,
+    handleChangeDepartment,
+    handleChangeNumber,
+    handleChangeGE,
+    handleSearch,
+    departmentCode,
+    number,
+    ge,
+    error,
+    geOptions,
+  } = useSearch();
+
+  useEffect(() => {
+    onUpdateCourses(courses);
+  }, [courses, onUpdateCourses]);
+
+  useEffect(() => {
+    onUpdateLoading(loading);
+  }, [loading, onUpdateLoading]);
+
+  useEffect(() => {
+    onUpdateLoadingMoreResults(loadingMoreResults);
+  }, [loadingMoreResults, onUpdateLoadingMoreResults]);
+
+  return (
+    <form
+      onSubmit={(event) => {
+        event.preventDefault();
+        handleSearch(departmentCode ?? "", number, ge ?? "");
+      }}
+    >
+      <div className="grid grid-cols-6 gap-2 p-2">
+        <Select
+          placeholder="Department"
+          name="department"
+          aria-label="department"
+          className="col-span-6"
+          variant="soft"
+          onChange={handleChangeDepartment}
+          value={departmentCode ?? ""}
+          size="md"
+          slotProps={{
+            listbox: {
+              sx: { minWidth: 270 },
+            },
+          }}
+        >
+          {departments.map((dep) => (
+            <Option key={dep.value} value={dep.value}>
+              {dep.label}
+            </Option>
+          ))}
+        </Select>
+        <Input
+          error={error}
+          className="w-full col-span-4"
+          color="neutral"
+          placeholder="Number"
+          variant="soft"
+          name="number"
+          aria-label="number"
+          onChange={(event) => handleChangeNumber(event.target.value)}
+          size="md"
+        />
+        <Select
+          placeholder="GE"
+          name="ge"
+          aria-label="ge"
+          className="col-span-2"
+          variant="soft"
+          onChange={handleChangeGE}
+          value={ge ?? ""}
+          size="md"
+        >
+          {geOptions.map((option) => (
+            <Option key={option.value} value={option.value}>
+              {option.label}
+            </Option>
+          ))}
+        </Select>
+        {error && (
```

Comment:
This helpful error message does not show up when inputting an invalid course number. Refactor it so it gives a clear error message if I mistype something like 'csehk 3a'